src/session_pool/session.js

```diff
@@ -264,8 +264,9 @@ export class Session {
      * @param {string} url
      */
     setPuppeteerCookies(cookies, url) {
+        const that = this;
         try {
-            this._setCookies(cookies.map(this._puppeteerCookieToTough), url);
+            this._setCookies(cookies.map(this._puppeteerCookieToTough.bind(that)), url);
```

Comment:
`this._puppeteerCookieToTough.bind(this)` will work

Reply:
Fixed.